### DISCUSSION TITLE
Custom class causing wrong error

### DISCUSSION THREAD

**Discussion by dirad** [tailwindlabs/tailwindcss#10185] [Help]
**What version of Tailwind CSS are you using?**

3.2.4

**What build tool (or framework if it abstracts the build tool) are you using?**

Astro 1.6.13

**What version of Node.js are you using?**

node 16.70

**What browser are you using?**

Brave / chrome / firefoxxxxx

**What operating system are you using?**

Windows

**Reproduction URL**

https://play.tailwindcss.com/LuN2ZO0DN0

**Describe your issue**

I've added a custom font to the site, and want to use it on occasion.
When making a simple class with the font name included, it works fine, but if I make a `@layer base` and create a real custom class, and then use `@apply font-class-name` things get strange.

I get an error saying that custom class doesnt exist.
Go to the declaration and delete the `@apply font-class-name` and save - the project loads fine
Go and put that rule in again - the site loads fine.

Now, if I stop the dev server, I'll get the error again when starting it.
To solve it I again delete, save, and undelete, save.

**Comment by adamwathan** [comment]
Hey! I've tried to simplify your demo because there was a lot of noise in there and stuff that didn't make sense to me (like using a `<style>` block in your HTML and trying to use Tailwind features like `@apply` there — that will never work) and things appear to work for me as expected:

https://play.tailwindcss.com/ofHfzDNdLB?file=css

Can you share something that includes the minimal changes to that demo to trigger a problem?

**Comment by dirad** [comment]
Thanks!

I included the mess because I worried it might have something  to do with my issue.

I actually have a huge `<style>` block in my HTML with plenty of `@apply`s, and it works great in my project... Why shouldn't it?  Is it because of the Tailwind Play system ?


In your cleaned demo I couldnt trigger the issue..

**Comment by adamwathan** [comment]
Are you using the [Play CDN](https://tailwindcss.com/docs/installation/play-cdn)? That's the only way that could maybe work, because Tailwind _compiles_ CSS files so it only actually processes external files, it doesn't process style blocks in your HTML.

Will need to actually see a real example project to give you a better answer here, going to convert this to a discussion though in the mean time.

**Reply by dirad** [reply]
Not using Play CDN.
I have an Astro dev, and deploy to Netlify.  Both work with compiling this CSS inside the block, and not only css files! 

So kind of confusing huh?   I have most of the styling going on in a style block in my `index.astro` file.
Maybe this is because of Astro somehow?
